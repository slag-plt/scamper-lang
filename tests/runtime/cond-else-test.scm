(import image)

(define factorial
  (lambda (n)
    (cond
      [(zero? n) 1]
      [else (* n (factorial (- n 1)))])))

(factorial 5)

(define red-square (rectangle 15 15 "solid" "red"))

(define type-of
  (lambda (datum)
    (cond
      [(number? datum) "number"]
      [(string? datum) "string"]
      [else "some-other-type"])))

(type-of red-square)

(cond
  [#f "shouldn't get here!"]
  [#f "oh no!"])