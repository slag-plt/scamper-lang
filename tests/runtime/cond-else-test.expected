120
"some-other-type"
